packages/coverage/src/coverage.ts

```diff
@@ -0,0 +1,269 @@
+import path from 'path';
+import fs from 'fs';
+import {
+  BettererDiff,
+  BettererRun,
+  BettererSerialiser,
+  BettererTest,
+  BettererTestOptionsComplex
+} from '@betterer/betterer';
+import { BettererConstraintResult } from '@betterer/constraints';
+import { BettererLog, BettererLogs } from '@betterer/logger';
+import {
+  BettererCoverageDiff,
+  BettererCoverageIssue,
+  BettererCoverageIssues,
+  BettererCoverageTest,
+  BettererCoverageTestOptions,
+  coverageAttributes,
+  IstanbulCoverageSummary,
+  IstanbulCoverageSummaryType,
+  IstanbulFileCoverage
+} from './types';
+import { reduceJoinRecords, uncoveredFile } from './helpers';
+import { isRight } from 'fp-ts/Either';
+import { BettererError } from '@betterer/errors';
+import { Validation } from 'io-ts';
+
+class BettererCoverageTestΩ
+  implements BettererTestOptionsComplex<BettererCoverageIssues, BettererCoverageIssues, BettererCoverageDiff>
+{
+  private readonly excludes: Array<RegExp>;
+  private readonly includes: Array<RegExp>;
+  private readonly coverageReportPath: string;
+  private readonly baseDir: string;
+  private readonly onlyTotalCoverage: boolean;
+
+  constructor(options: BettererCoverageTestOptions) {
+    this.baseDir = path.resolve(options.baseDir);
+    this.coverageReportPath = path.resolve(this.baseDir, options.coverageFile);
+    this.excludes = options.excludeFiles;
+    this.includes = options.includeFiles;
+    this.onlyTotalCoverage = options.totalCoverage;
+  }
+
+  public differ(
+    expectedIssues: BettererCoverageIssues,
+    actualIssues: BettererCoverageIssues
+  ): BettererDiff<BettererCoverageDiff> {
+    const logs: BettererLogs = [];
+    const diff: BettererCoverageDiff = {};
+    if (Object.keys(actualIssues).length === 0) {
+      logs.push({ warn: 'The result is empty. Have you maybe forgotten to run your tests?' });
+    } else {
+      Object.assign(diff, this.detectNewOrUpdatedFiles(actualIssues, expectedIssues, logs));
+    }
+    Object.assign(diff, this.detectRemovedFiles(expectedIssues, actualIssues, logs));
+    return {
+      diff,
+      logs
+    };
+  }
+
+  public constraint(
+    actualIssues: BettererCoverageIssues,
+    expectedIssues: BettererCoverageIssues
+  ): BettererConstraintResult {
+    const { diff } = this.differ(expectedIssues, actualIssues);
+    const { isWorse, isImproved } = this.isWorseOrImproved(diff);
+    if (isWorse) {
+      return BettererConstraintResult.worse;
+    }
+    if (isImproved) {
+      return BettererConstraintResult.better;
+    }
+    return BettererConstraintResult.same;
+  }
+
+  public goal(result: BettererCoverageIssues): boolean {
+    return Object.keys(result).every((filePath) =>
+      coverageAttributes.every((attribute) => result[filePath][attribute] === 0)
+    );
+  }
+
+  public get serialiser(): BettererSerialiser<BettererCoverageIssues, BettererCoverageIssues> {
```

Comment:
The default `serialise` and `deserialise` functions are just pass throughs, so you shouldn't need these

Reply:
BettereTestOptionsComplex requires that I implement these myself.